### ISSUE TITLE
Chef client scheduled task on windows not compatible with chef 14

### ISSUE THREAD

**Issue by scotthain** [chef-cookbooks/chef-client#565]
### Cookbook version
10.0.2

### Chef-client version
[Version of chef-client in your environment]

### Platform Details
[Operating system distribution and release version. Cloud provider if running in the cloud]

### Scenario:
Use `chef_client_scheduled_task` on windows 2012r2 (ec2, latest AMI) with chef 14.0.168 - 

### Steps to Reproduce:
[If you are filing an issue what are the things we need to do in order to repro your problem? How are you using this cookbook or any resources it includes?]

### Expected Result:
[What are you expecting to happen as the consequence of above reproduction steps?]

### Actual Result:
```
       Recipe: chef-client::task
         * windows_service[chef-client] action configure_startup (skipped due to only_if)
         * chef_client_scheduled_task[Chef Client] action add

           ================================================================================
           Error executing action `add` on resource 'chef_client_scheduled_task[Chef Client]'
           ================================================================================

           NoMethodError
           -------------
           undefined method `resources' for #<#<Class:0x0000000005f381d0>:0x0000000009eb4818>
           Did you mean?  find_resource

           Resource Declaration:
           ---------------------
           # In C:/Users/ADMINI~1/AppData/Local/Temp/kitchen/cache/cookbooks/chef-client/recipes/task.rb

            37: chef_client_scheduled_task 'Chef Client' do
            38:   user node['chef_client']['task']['user']
            39:   password node['chef_client']['task']['password']
            40:   frequency node['chef_client']['task']['frequency']
            41:   frequency_modifier lazy { node['chef_client']['task']['frequency_modifier'] }
            42:   start_time node['chef_client']['task']['start_time']
            43:   start_date node['chef_client']['task']['start_date']
            44:   splay node['chef_client']['splay']
            45:   config_directory node['chef_client']['conf_dir']
            46:   log_directory node['chef_client']['log_dir']
            47:   chef_binary_path node['chef_client']['bin']
            48:   daemon_options node['chef_client']['daemon_options']
            49: end

           Compiled Resource:
           ------------------
           # Declared in C:/Users/ADMINI~1/AppData/Local/Temp/kitchen/cache/cookbooks/chef-client/recipes/task.rb:37:in `from_file'

           chef_client_scheduled_task("Chef Client") do
             action [:add]
             default_guard_interpreter :default
             declared_type :chef_client_scheduled_task
             cookbook_name "chef-client"
             recipe_name "task"
             user "SYSTEM"
             password nil
             frequency "minute"
             frequency_modifier 30
             start_time nil
             start_date nil
             splay 60
             config_directory "C:/chef"
             log_directory "C:/chef/log"
             chef_binary_path "C:\\opscode\\chef\\bin\\chef-client.bat"
             daemon_options []
           end

           System Info:
           ------------
           chef_version=14.0.168
           platform=windows
           platform_version=6.2.9200
           ruby=ruby 2.5.0p0 (2017-12-25 revision 61468) [x64-mingw32]
           program_name=C:/opscode/chef/bin/chef-client
           executable=C:/opscode/chef/bin/chef-client

**Comment by mccw** [comment]
I'm seeing the same issue with version 14.0.190.

**Comment by markaxon** [comment]
Seeing same issue on 14.0.190 windows server 2012r2 and 2016

**Comment by mccw** [comment]
Since this broke all of my nodes, I ended up deleting chef-client and creating a recipe using windows_task to schedule the chef-client run every hour. This is basically all I was using the chef-client cookbook for anyway.

**Comment by xuxogarcia** [comment]
This just broke me also :(

**Comment by wagostin** [comment]
The same issue. Any idea when this will be fixed?

**Comment by wamoore** [comment]
We've got the same issue.  Any word on a fix?

**Comment by tas50** [comment]
The fix is a new build of Chef 14 that we're working to release. We have the fix in master, but we need to wrap up a few other changes and get a 14.0.x release out.

**Comment by tas50** [comment]
A workaround has been released in 10.0.3 and the bug that caused this issue in Chef 14 will go out next week
